drgn_tools/cpuinfo.py

```diff
@@ -0,0 +1,615 @@
+            if check_smt_enabled(prog):
+                mitigation += "vulnerable"
+            else:
+                mitigation += "disabled"
+
+    return mitigation
+
+
+def get_mds_mitigation(prog: Program, cpu_caps_bugs: Object) -> str:
+    """
+    Extracts Mitigation for MDS
+    """
+    mitigation = (
+        prog["mds_strings"][prog["mds_mitigation"]].string_().decode("utf-8")
+    )
+    mitigation += "; SMT "
+    nbugints = 1
+    ncapints = len(cpu_caps_bugs) - nbugints
+    if (cpu_caps_bugs[4] >> 31) & 1:
+        mitigation += "Host state unknown"
+    elif (cpu_caps_bugs[ncapints] >> 20) & 1:
+        if prog["mds_mitigation"] == prog["MDS_MITIGATION_OFF"]:
+            mitigation += "vulnerable"
+        elif check_smt_enabled(prog):
+            mitigation += "mitigated"
+        else:
+            mitigation += "disabled"
+    else:
+        if check_smt_enabled(prog):
+            mitigation += "vulnerable"
+        else:
+            mitigation += "disabled"
+
+    return mitigation
+
+
+def get_taa_mitigation(prog: Program, cpu_caps_bugs: Object) -> str:
+    """
+    Extracts Mitigation for tsx_async_abort
+    """
+    mitigation = (
+        prog["taa_strings"][prog["taa_mitigation"]].string_().decode("utf-8")
+    )
+
+    if (
+        prog["taa_mitigation"] == prog["TAA_MITIGATION_TSX_DISABLED"]
+        or prog["taa_mitigation"] == prog["TAA_MITIGATION_OFF"]
+    ):
+        pass
+    elif (cpu_caps_bugs[4] >> 31) & 1:
+        mitigation += "; SMT Host state unknown"
+    else:
+        mitigation += "; SMT "
+        if check_smt_enabled(prog):
+            mitigation += "vulnerable"
+        else:
+            mitigation += "disabled"
+
+    return mitigation
+
+
+def get_itlb_multihit_mitigation(prog: Program, cpu_caps_bugs: Object) -> str:
+    """
+    Extracts Mitigation for itlb_multihit
+    """
+    if "l1tf_vmx_states" not in prog:
+        mitigation = "Processor vulnerable"
+    elif "cr4_read_shadow" in prog:
+        if (not ((cpu_caps_bugs[7] >> 31) & 1)) or (
+            not ((cpu_caps_bugs[4] >> 5) & 1)
+        ):
+            mitigation = "KVM: Mitigation: VMX unsupported"
+        elif not per_cpu(prog["cpu_tlbstate"], 1).cr4 & (1 << 13):
+            mitigation = "KVM: Mitigation: VMX disabled"
+    elif prog["itlb_multihit_kvm_mitigation"]:
+        mitigation = "KVM: Mitigation: Split huge pages"
+    else:
+        mitigation = "KVM: Vulnerable"
+
+    return mitigation
+
+
+def get_srbds_mitigation(prog: Program) -> str:
+    """
+    Extracts Mitigation for SRBDS
+    """
+    return (
+        prog["srbds_strings"][prog["srbds_mitigation"]]
+        .string_()
+        .decode("utf-8")
+    )
+
+
+def get_mmio_stale_data_mitigation(
+    prog: Program, cpu_caps_bugs: Object
+) -> str:
+    """
+    Extracts Mitigation for mmio_stale_data and mmio_unknown
+    """
+    nbugints = 1
+    ncapints = len(cpu_caps_bugs) - nbugints
+    if (cpu_caps_bugs[ncapints] >> 26) & 1:
```

Comment:
constant + test_bit

Since this is a "bug" field, which means you need to determine "ncapints" first, be sure to detect ncapints correctly via `len(prog["x86_cap_flags"]) / 32`, rather than hard-coding nbugints to 1 as you have here.